src/NuGet.Clients/NuGet.VisualStudio.Common/NuGetFeedbackDiagnosticFileProvider.cs

```diff
@@ -1,24 +1,193 @@
     {
+        // Don't crash VS feedback if imports fail. Maybe broken MEF is exactly what the customer is trying to report!
+        [Import(AllowDefault = true)]
+        public ISolutionManager? SolutionManager { get; set; }
+
+        [Import(AllowDefault = true)]
+        public INuGetTelemetryProvider? TelemetryProvider { get; set; }
+
+        [Import(AllowDefault = true)]
+        public ISettings? Settings { get; set; }
+
+        /// <summary>Used for testing, so tests can wait for the background task to finish.</summary>
+        public event EventHandler<Task>? BackgroundTaskStarted;
+
         public IReadOnlyCollection<string> GetFiles()
         {
-            // todo: https://github.com/NuGet/Home/issues/8605
+#pragma warning disable CA2000 // Dispose objects before losing scope
+            // fileStream gets disposed in the background task below.
+            FileStream fileStream = GetOutputFile();
+#pragma warning restore CA2000 // Dispose objects before losing scope
+
+            string fileName = fileStream.Name;
+
+            // See comments on IFeedbackDiagnosticFileProvider.GetFiles
+            Task task = Task.Run(() => WriteToZipAndCloseFileAsync(fileStream));
+            BackgroundTaskStarted?.Invoke(this, task);
 
             return new List<string>()
             {
-                DependencyGraphRestoreUtility.GetDefaultDGSpecFileName()
+                fileName
             };
         }
+
+        private async Task WriteToZipAndCloseFileAsync(Stream stream)
+        {
+            using (stream)
+            {
+                try
+                {
+                    await WriteToZipAsync(stream);
+                }
+                catch (Exception exception)
+                {
+                    await PostFaultAsync(exception);
+                }
+            }
+        }
+
+        private FileStream GetOutputFile()
+        {
+            string fileNamePrefix = "nuget." + DateTime.UtcNow.ToString("yyyy-MM-dd.HH-mm-ss");
+            for (int attempt = 0; attempt < 100; attempt++)
+            {
+                string fileName = attempt == 0
+                    ? fileNamePrefix + ".zip"
+                    : fileNamePrefix + "." + attempt + ".zip";
+                string fullPath = Path.Combine(Path.GetTempPath(), fileName);
+                try
+                {
+                    FileStream fileStream = new FileStream(fullPath, FileMode.CreateNew);
+                    return fileStream;
+                }
+                catch (IOException)
+                {
+                    // Unlikely that the customer is going to click "report a problem" more than once per second, but just in case
+                }
+            }
+
+            throw new Exception("Unable to create file to attach");
+        }
+
+        public async Task WriteToZipAsync(Stream stream)
+        {
+            var telemetry = new TelemetryEvent("feedback");
```

Comment:
Maybe `vsfeedback` would be more appropriate?

Reply:
the full name, after our infrastructure prepends the static part, is `vs/nuget/feedback`. adding in another `vs` feels redundant to me. 🤷